ui/lib/services/bom_bar_client.dart

```diff
@@ -38,9 +38,10 @@ class BomBarClient {
     ));
   }
 
-  /// Returns all available projects.
-  Future<List<Project>> getProjects() async {
-    final response = await dio.getUri(_projectsUrl);
+  /// Returns all or projects matching the provided [filter] fragment.
+  Future<List<Project>> findProjectsBySearchFragment(String? filter) async {
```

Comment:
This function is "searching by search fragment", but then accepting a parameter called `filter`.

Suggestion is to name the function e.g. `findProjects()` and naming the parameter something like "fragment".